pkg/auth/rbac_test.go

```diff
@@ -0,0 +1,2012 @@
+					SrcIp: []byte{192, 168, 122, 4},
+				},
+			},
+			true,
+		},
+		{
+			"9-2. Test matches OR-ed allow, no match matches, deny",
+			fields{
+				&policyStore{
+					byKey: map[string]*authPolicy{
+						ALLOW_POLICY: {
+							&security.Authorization{
+								Name:      ALLOW_AUTH,
+								Namespace: KMESH_NAMESPACE,
+								Scope:     security.Scope_WORKLOAD_SELECTOR,
+								Action:    security.Action_ALLOW,
+								Rules: []*security.Rule{
+									{
+										Clauses: []*security.Clause{
+											{
+												Matches: []*security.Match{
+													{
+														DestinationIps: []*security.Address{
+															{
+																Address: []byte{192, 168, 122, 3},
+																Length:  32,
+															},
+														},
+													},
+													{
+														DestinationIps: []*security.Address{
+															{
+																Address: []byte{192, 168, 122, 4},
+																Length:  32,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					byNamespace: map[string]map[string]struct{}{
+						KMESH_NAMESPACE: {ALLOW_POLICY: struct{}{}},
+					},
+				},
+				&workloadapi.Workload{Namespace: KMESH_NAMESPACE},
+			},
+			args{
+				&RbacConnection{
+					DstIp: []byte{192, 168, 122, 2},
+					SrcIp: []byte{192, 168, 122, 5},
+				},
+			},
+			false,
+		},
+		{
+			"9-3. Test matches OR-ed deny, 1 match matches, deny",
+			fields{
+				&policyStore{
+					byKey: map[string]*authPolicy{
+						DENY_POLICY: {
+							&security.Authorization{
+								Name:      DENY_AUTH,
+								Namespace: KMESH_NAMESPACE,
+								Scope:     security.Scope_WORKLOAD_SELECTOR,
+								Action:    security.Action_DENY,
+								Rules: []*security.Rule{
+									{
+										Clauses: []*security.Clause{
+											{
+												Matches: []*security.Match{
+													{
+														NotDestinationIps: []*security.Address{
+															{
+																Address: []byte{192, 168, 122, 2},
+																Length:  32,
+															},
+														},
+													},
+													{
+														NotDestinationIps: []*security.Address{
+															{
+																Address: []byte{192, 168, 122, 3},
+																Length:  32,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					byNamespace: map[string]map[string]struct{}{
+						KMESH_NAMESPACE: {DENY_POLICY: struct{}{}},
+					},
+				},
+				&workloadapi.Workload{Namespace: KMESH_NAMESPACE},
+			},
+			args{
+				&RbacConnection{
+					DstIp: []byte{192, 168, 122, 2},
+					SrcIp: []byte{192, 168, 122, 4},
+				},
+			},
+			false,
+		},
+		{
+			"9-4. Test matches OR-ed deny, no match matches, allow",
+			fields{
+				&policyStore{
+					byKey: map[string]*authPolicy{
+						DENY_POLICY: {
+							&security.Authorization{
```

Comment:
nit: define the auth policy beforehand, so we donot need to inline a whole policy in each case. It is too long now

Reply:
done